tests/unit-tests/crud/data.php

```diff
@@ -273,4 +273,48 @@ function test_add_meta_data_overwrite_before_save() {
 		$object->update_meta_data( 'test_field_0', 'another field 2' );
 		$this->assertEquals( 'val1', $object->get_meta( 'test_field_2' ) );
 	}
+
+	/**
+	 * Test protected method set_date_prop by testing a order date setter.
+	 */
+	function test_set_date_prop() {
+		$object = new WC_Order();
+
+		// Change timezone in WP.
+		update_option( 'gmt_offset', -4 );
```

Comment:
And if you really want to check how robust the new date handling is, add another two rounds of tests to run after:

* `date_default_timezone_set( 'Pacific/Fiji' )` is called, and then
* `date_default_timezone_set( 'Pacific/Tahiti' )` is called.

It might be beyond the scope of #13498, but WordPress expects the default timezone to be `UTC`, because it sets it to that, but some APIs in PHP, like `date()` or `strtotime()` have different behaviour depending on the default timezone set, which can lead to different date/timestamp values when 3rd party code is calling `date_default_timezone_set()` with a timezone other than `UTC` (to see more background on this issue, see https://github.com/Prospress/woocommerce-subscriptions/pull/1102 or https://github.com/Prospress/woocommerce-subscriptions/pull/892).

I'm not 100% sure, but I think the uses of `date()` here may be effected when `date_default_timezone_set()` is set for a timezone other than `UTC` (although IIRC, it was only an issue when the 2nd timestamp param wasn't set). The tests will help check for that.

We run [Action Scheduler tests](https://github.com/Prospress/action-scheduler/pull/43/commits/2e5810a7573e7cc32f9aa4433aaa4fcec7f75ee4) under 3 different default timezone settings for this reason.